tests/unit/test_solvers.py

```diff
@@ -850,6 +850,34 @@ def test_base_solver(
             solver_factory.keywords["base_solver"] = base_solver
             solver_factory()
 
+    def test_map_reduce_diverse_selection(self):
+        """Check if MapReduce returns indices from multiple partitions."""
+        dataset_size = 40
+        data_dim = 5
+        coreset_size = 6
+        leaf_size = 12
+
+        key = jr.PRNGKey(0)
+        dataset = jr.normal(key, shape=(dataset_size, data_dim))
+
+        kernel = SquaredExponentialKernel()
+        base_solver = KernelHerding(coreset_size=coreset_size, kernel=kernel)
```

Comment:
@tc85324 For the sake of minimising overhead, and since we both already have visibility on this PR, I’m happy for the changes that rework the existing tests to be made here.

I don’t think it warrants creating a new issue and PR.